docs/docs/cloud/how-tos/remote_graph.md

```diff
@@ -0,0 +1,150 @@
+
+
+### Using URL
+
+=== "Python"
+
+    ```python
+    from langgraph.pregel.remote import RemoteGraph
+
+    url = <DEPLOYMENT_URL>
+    graph_name = "agent"
+    remote_graph = RemoteGraph(graph_name, url=url)
+    ```
+
+### Using clients
+
+=== "Python"
+
+    ```python
+    from langgraph_sdk import get_client, get_sync_client
+    from langgraph.pregel.remote import RemoteGraph
+
+    url = <DEPLOYMENT_URL>
+    graph_name = "agent"
+    client = get_client(url=url)
+    sync_client = get_sync_client(url=url)
+    remote_graph = RemoteGraph(graph_name, client=client, sync_client=sync_client)
+    ```
+
+## Invoking the graph
+
+Since `RemoteGraph` is a `Runnable` that implements the same methods as `CompiledGraph`, you can interact with it the same way you normally would with a compiled graph, i.e. by calling `.invoke()`, `.stream()`, `.get_state()`, `.update_state()`, etc (as well as their async counterparts).
+
+### Asynchronously
+
+!!! Note
+
+    To use the graph asynchronously, you must provide either the `url` or `client` when initializing the `RemoteGraph`.
+
+=== "Python"
+
+    ```python
+    # invoke the graph
+    result = await remote_graph.ainvoke({
+        "messages": [{"role": "user", "content": "what's the weather in sf"}]
+    })
+
+    # stream outputs from the graph
+    async for chunk in remote_graph.astream({
+        "messages": [("user", "what's the weather in la?")]
+    }):
+        print(chunk)
+    ```
+
+### Synchronously
+
+!!! Note
+
+    To use the graph synchronously, you must provide either the `url` or `sync_client` when initializing the `RemoteGraph`.
+
+=== "Python"
+
+    ```python
+    # invoke the graph
+    result = remote_graph.invoke({
+        "messages": [{"role": "user", "content": "what's the weather in sf"}]
+    })
+
+    # stream outputs from the graph
+    for chunk in remote_graph.stream({
+        "messages": [("user", "what's the weather in la?")]
+    }):
+        print(chunk)
+    ```
+
+## Thread-level persistence
+
+By default, the graph runs (i.e. `.invoke()` or `.stream()` invocations) are stateless - the checkpoints and the final state of the graph are not persisted. If you would like to persist the outputs of the graph run (for example, to enable human-in-the-loop features), you can create a thread and provide the thread ID via the `config` argument, same as you would with a regular compiled graph:
+
+=== "Python"
+
+    ```python
+    from langgraph_sdk import get_sync_client
+    url = <DEPLOYMENT_URL>
+    graph_name = "agent"
+    sync_client = get_sync_client(url=url)
+    remote_graph = RemoteGraph(graph_name, url=url)
+
+    # create a thread (or use an existing thread instead)
+    thread = sync_client.threads.create()
+
+    # invoke the graph with the thread config
+    config = {"configurable": {"thread_id": thread["thread_id"]}}
```

Comment:
any issues with thread_id if used as a sub-graph and the thread_id is provided top level? Any possibility to end up in an inconsistent state?

Reply:
good question, let me double check